In [154]:
import numpy as np
import pandas as pd
import datetime

In [155]:
p1_gen = pd.read_csv("Plant_1/Plant_1_Generation_Data.csv")
p1_weath = pd.read_csv("Plant_1/Plant_1_Weather_Sensor_Data.csv")

p2_gen = pd.read_csv("Plant_2/Plant_2_Generation_Data.csv")
p2_weath = pd.read_csv("Plant_2/Plant_2_Weather_Sensor_Data.csv")

In [151]:
# CONTROL - all dates are present
dff = df_2g
dff['DATE'] = pd.to_datetime(dff['DATE_TIME'], format="%d-%m-%Y %H:%M").dt.time
#print(dff['DATE'].value_counts()) # 49 of different measured time moments
# count how many rows are missing
print(49*748 - dff['DATE'].value_counts().sum()) # if generation
#print(49*34 - dff['DATE'].value_counts().sum()) #if weather

2146


In [4]:
#dff[ (dff['DATE_TIME'].dt.date == datetime.date(2020, 10,6) )]

In [156]:
for df in [p1_gen]:
    df['DATE_TIME'] = pd.to_datetime(df['DATE_TIME'], format="%d-%m-%Y %H:%M")
    
for df in [p1_weath, p2_gen, p2_weath]:
    df['DATE_TIME'] = pd.to_datetime(df['DATE_TIME'], format="%Y-%m-%d %H:%M:%S")

In [157]:
df_1g = p1_gen.copy()
df_1w = p1_weath.copy()
df_2g = p2_gen.copy()
df_2w = p2_weath.copy()

#for df in [df_1g, df_1w, df_2g, df_2w]:
df_1g = df_1g[ (df_1g['DATE_TIME'].dt.time >= datetime.time(6, 0) ) & (df_1g['DATE_TIME'].dt.time <= datetime.time(18, 0)) ]
df_1w = df_1w[ (df_1w['DATE_TIME'].dt.time >= datetime.time(6, 0) ) & (df_1w['DATE_TIME'].dt.time <= datetime.time(18, 0)) ]

df_2g = df_2g[ (df_2g['DATE_TIME'].dt.time >= datetime.time(6, 0) ) & (df_2g['DATE_TIME'].dt.time <= datetime.time(18, 0)) ]
df_2w = df_2w[ (df_2w['DATE_TIME'].dt.time >= datetime.time(6, 0) ) & (df_2w['DATE_TIME'].dt.time <= datetime.time(18, 0)) ]

In [158]:
p1_gen

DATE_TIME  PLANT_ID       SOURCE_KEY  DC_POWER  AC_POWER  \
0     2020-05-15 00:00:00   4135001  1BY6WEcLGh8j5v7       0.0       0.0   
1     2020-05-15 00:00:00   4135001  1IF53ai7Xc0U56Y       0.0       0.0   
2     2020-05-15 00:00:00   4135001  3PZuoBAID5Wc2HD       0.0       0.0   
3     2020-05-15 00:00:00   4135001  7JYdWkrLSPkdwr4       0.0       0.0   
4     2020-05-15 00:00:00   4135001  McdE0feGgRqW7Ca       0.0       0.0   
...                   ...       ...              ...       ...       ...   
68773 2020-06-17 23:45:00   4135001  uHbuxQJl8lW7ozc       0.0       0.0   
68774 2020-06-17 23:45:00   4135001  wCURE6d3bPkepu2       0.0       0.0   
68775 2020-06-17 23:45:00   4135001  z9Y9gH1T5YWrNuG       0.0       0.0   
68776 2020-06-17 23:45:00   4135001  zBIq5rxdHJRwDNY       0.0       0.0   
68777 2020-06-17 23:45:00   4135001  zVJPv84UY57bAof       0.0       0.0   

       DAILY_YIELD  TOTAL_YIELD  
0            0.000    6259559.0  
1            0.000    6183645.0  
2            0.000    6987759.0  
3            0.000    7602960.0  
4            0.000    7158964.0  
...            ...          ...  
68773     5967.000    7287002.0  
68774     5147.625    7028601.0  
68775     5819.000    7251204.0  
68776     5817.000    6583369.0  
68777     5910.000    7363272.0  

[68778 rows x 7 columns]

GENERATION DATA SHOULD BE 71 808 ROWS!!  (4x24x34x22)
Plant 1 generation - 68 778 (missing 3030)
daytime: 764
Plant 2 generation - 67 698 (missing 4110)
daytime: 2146

WEATHER DATA SHOULD BE 3264 ROWS!!  (4x24x34)
Plant 1 weather - 3182 (missing 82)
daytime: 31
Plant 2 weather - 3259 (missing 5)
daytime: 4

# ON ALL DATA (including night-time)

In [165]:
# p1_gen, p1_weath, p2_gen, p2_weath

dc = []

gen_zeros = []
for df in [p1_gen, p2_gen]:
    DC = (df['DC_POWER'] > 0).sum()
    AC = (df['AC_POWER'] > 0).sum()
    DY = (df['DAILY_YIELD'] > 0).sum()
    TY = (df['TOTAL_YIELD'] > 0).sum()
    nonzeros = DC + AC + DY + TY
    zeros = len(df)*4 - nonzeros
    gen_zeros.append(zeros)
    dc.append( len(df) - DC )
    
gen_zeros[0] = gen_zeros[0] + 3030*4
gen_zeros[1] = gen_zeros[1] + 4110*4
dc[0] = dc[0] + 3030
dc[1] = dc[1] + 4110
    
weath_zeros = []
for df in [p1_weath, p2_weath]:
    amb = (df['AMBIENT_TEMPERATURE'] > 0).sum()
    mod = (df['MODULE_TEMPERATURE'] > 0).sum()
    irr = (df['IRRADIATION'] > 0).sum()
    nonzeros = amb + mod + irr
    zeros = len(df)*3 - nonzeros
    weath_zeros.append(zeros)
    
weath_zeros[0] = weath_zeros[0] + 82*3
weath_zeros[1] = weath_zeros[1] + 5*3

In [166]:
print(dc)
print(dc[0]/71808)
print(dc[1]/71808)

print(gen_zeros)
print(weath_zeros)
print((gen_zeros[0]+weath_zeros[0])/(71808*4+3264*3))
print((gen_zeros[1]+weath_zeros[1])/(71808*4+3264*3))

[34981, 39772]
0.4871462789661319
0.5538658645276292
[94718, 99896]
[1671, 1412]
0.3245158640379229
0.34107681534152123


In [167]:
#p1_gen[p1_gen['DC_POWER']>0]['DC_POWER'].size #ON VÕRDNE SELLEGA (p1_gen['DC_POWER'] > 0).sum()

# ON ONLY DAYTIME DATA

In [162]:
dc = []

gen_zeros = []
for df in [df_1g, df_2g]:
    DC = (df['DC_POWER'] > 0).sum()
    AC = (df['AC_POWER'] > 0).sum()
    DY = (df['DAILY_YIELD'] > 0).sum()
    TY = (df['TOTAL_YIELD'] > 0).sum()
    nonzeros = DC + AC + DY + TY
    zeros = len(df)*4 - nonzeros
    gen_zeros.append(zeros)
    dc.append( len(df)-DC )

gen_zeros[0] = gen_zeros[0] + 764*4
gen_zeros[1] = gen_zeros[1] + 2146*4
dc[0] = dc[0] + 764
dc[1] = dc[1] + 2146

weath_zeros = []
for df in [df_1w, df_2w]:
    amb = (df['AMBIENT_TEMPERATURE'] > 0).sum()
    mod = (df['MODULE_TEMPERATURE'] > 0).sum()
    irr = (df['IRRADIATION'] > 0).sum()
    nonzeros = amb + mod + irr
    zeros = len(df)*3 - nonzeros
    weath_zeros.append(zeros)
    
weath_zeros[0] = weath_zeros[0] + 31*3
weath_zeros[1] = weath_zeros[1] + 4*3

In [164]:
print('DC')
print(dc)
print(dc[0]/( len(df_1g)+764) )
print(dc[1]/( len(df_2g)+2146) )

print('ALL')
print(gen_zeros)
print(weath_zeros)
print((gen_zeros[0]+weath_zeros[0])/( (len(df_1g)+764)*4 + (len(df_1w)+31)*3 ))
print((gen_zeros[1]+weath_zeros[1])/( (len(df_2g)+2146)*4 + (len(df_2w)+4)*3 ))

DC
[981, 6149]
0.02676525155516752
0.1677671068427371
ALL
[3765, 17738]
[93, 13]
0.02544754165402425
0.11708639499755946


##### Plant 1 zeros
Gen = 709, all = 143 552

Weather = 0, all = 4905

Together = 709/148 457 = 0.005 -> 0.5%

DC = 217/35 888 = 0.006 -> 0.6% (217/709 -> 30% from all missing values)

##### Plant 2 zeros
Gen = 9154, all = 138 024

Weather = 1, all = 4986

Together = 9155/143 010 = 0.064 -> 6.4%

DC = 4003/34 506 = 0.116 -> 11.6 or 12% (4003/9154 -> 44% of all missing values)